text/0757-default-modifier-manager.md

```diff
@@ -0,0 +1,276 @@
+RFC PR: https://github.com/emberjs/rfcs/pull/757
+---
+
+# Default Modifier Manager
+
+## Summary
+
+Anything that can be in a template has its lifecycle managed by the manager pattern.
+Today, we have known managers for `@glimmer/component`, `@ember/helper`, etc.
+But what happens when the VM encounters an object for which there is no manager?,
+such as a plain function? This RFC explores and proposes a default behavior for
+those unknown scenarios when it comes to _modifiers_.
+
+## Motivation
+
+The addon, [ember-could-get-used-to-this](gh-ecgutt)
+demonstrated that it's possible to (almost) use plain functions for modifiers.
+And since Ember 3.25, modifiers can be passed around as values -- which has opened
+a whole new world of ergonomics improvements where a developer could define a function in a
+component class and use that function **as** a modifier. This is thanks to
+ember-could-get-used-to-this implementing a Modifier Manager that knows what to do with plain functions.
+
+This has the impact of greatly reducing the mental burden for app and addon devs, in that,
+they no longer need to jump over to the app/addon modifiers directory to create a modifier
+"just to do this one simple thing". In Ember 3.25+, developers can now do `<div {{this.myModifier}}>`.
+
+The introduction of a plain-function modifier-manager is important because over the past recent years
+(since the introduction of the Octane Edition), we've seen on numerous occasions,
+folks repeatedly reaching for `{{did-insert}}`, `{{did-update}}`,
+(from [`@ember/render-modifiers`](gh-render-modifiers)) and other lifecycle-esque utilities.
+
+The README of [ember-modifier][gh-ember-modifier] does an excellent job of describing side-effects,
+modifiers, how to think about them, and there overall philosophy.
+
+There are two excerpts from the modifier addons' READMEs to keep in mind in the rest of this RFC,
+
+> Conceptually, modifiers take tracked, derived state, and turn it into some sort of side effect - usually, mutating the DOM node they are applied to in some way, but they might also trigger other types of side effects.
+>
+> <cite style="display: block; text-align: right;">ember-modifier's README</site>
+
+and
+
+> In many cases, classic lifecycle hooks like didInsertElement can be rewritten as custom modifiers that internalize functionality manipulating or generating state from a DOM element. Other times, you may find that a modifier is not the right fit for that logic at all, in which case it's worth revisiting the design to find a better pattern.
+>
+> <cite style="display: block; text-align: right;">@ember/render-modifier's README</site>
+
+
+An example:
+
+```js
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+
+export default class Example extends Component {
+  @tracked numberOfClicks = 0
+
+  // when the number of clicks is odd, add a JS movement effect / animation
+  wiggly = (element) => {
```

Comment:
It'd be good to call out that this is the modifier with something like `// MODIFIER DEFINITION` or something.

Reply:
added